multisrc/overrides/animestream/mykdrama/src/MyKdrama.kt

```diff
@@ -0,0 +1,121 @@
+import eu.kanade.tachiyomi.multisrc.animestream.AnimeStreamFilters
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.util.asJsoup
+import eu.kanade.tachiyomi.util.parallelCatchingFlatMapBlocking
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class MyKdrama : AnimeStream(
+    "fr",
+    "MyKdrama",
+    "https://mykdrama.co",
+) {
+    override val animeListUrl = "$baseUrl/drama"
+
+    override val dateFormatter by lazy {
+        SimpleDateFormat("MMMM dd, yyyy", Locale("fr"))
+    }
+
+    // =============================== Search ===============================
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val params = getSearchParameters(filters)
+        return if (query.isNotEmpty()) {
+            GET("$baseUrl/page/$page/?s=$query")
+        } else {
+            val queryParams = params.run { listOf(genres) }
+                .filter(String::isNotBlank)
+                .joinToString("&")
+            val url = "$animeListUrl/?$queryParams".toHttpUrl().newBuilder()
+                .addQueryParameter("page", "$page")
+                .addIfNotBlank("status", params.status)
+                .addIfNotBlank("type", params.type)
+                .addIfNotBlank("order", params.order)
+                .build()
+            GET(url.toString(), headers)
+        }
+    }
+
+    // ============================== Filters ===============================
+    override val filtersSelector = "div.filter > ul"
+
+    override fun getFilterList(): AnimeFilterList {
+        return if (AnimeStreamFilters.filterInitialized()) {
+            AnimeFilterList(
+                GenresFilter("Genres"),
+                CountryFilter("Pays"),
+                AnimeFilter.Separator(),
+                StatusFilter("Status"),
+                TypeFilter("Type"),
+                OrderFilter("Ordre"),
+            )
+        } else {
+            AnimeFilterList(AnimeFilter.Header(filtersMissingWarning))
+        }
+    }
+
+    // =========================== Anime Details ============================
+
+    // ============================== Episodes ==============================
+
+    // ============================ Video Links =============================
+    override val prefQualityValues = arrayOf("1080p", "720p", "480p", "360p", "240p", "144p")
+    override val prefQualityEntries = prefQualityValues
+
+    override fun videoListParse(response: Response): List<Video> {
+        val doc = response.use { it.asJsoup() }
+        return doc.select(videoListSelector()).parallelCatchingFlatMapBlocking { element ->
+            val name = element.text()
+            val url = getHosterUrl(element)
+            getVideoList(url, name)
+        }.ifEmpty {
+            return doc.select(".gov-the-embed").parallelCatchingFlatMapBlocking { element ->
```

Comment:
Is using `return` twice really necessary?

Reply:
> Is using `return` twice really necessary?

Holy shit "twice" like the kpop group??